### ISSUE TITLE
Upgrade to Ash 3.0

### ISSUE THREAD

**Issue by wintermeyer** [wintermeyer/animina#431]
I have no idea how much work this is. Please add some information to this issue if we are talking more than a day before starting it.

**Comment by MICHAELMUNAVU83** [comment]
Starting on this task now , I will use the official docs and provide updates and estimations of how much time this will take https://hexdocs.pm/ash/upgrading-to-3-0.html

**Comment by MICHAELMUNAVU83** [comment]
The tasks are 


- [ ] Rename Ash Api to Ash.Domain
-To make this change you will need to do two things:

1. replace Ash.Api with [Ash.Domain](https://hexdocs.pm/ash/Ash.Domain.html) in your application
2. replace places where an :api option is passed to a function with the :domain option. For example, AshPhoenix.Form.for_create(..., api: MyApp.SomeApi) should now be AshPhoenix.Form.for_create(..., domain: MyApp.SomeDomain)
3. Update your application config to define ash_domains instead of ash_apis, eg. config :my_app, ash_domains: [MyApp.MyDomain]

**Comment by wintermeyer** [comment]
@MICHAELMUNAVU83 Is this still in progress or did you hit a road block?

**Comment by MICHAELMUNAVU83** [comment]
It is quite heavy , will take a lot of time , I was working on this before you were to pitch then I moved on to more pressing issues

**Comment by wintermeyer** [comment]
Can you estimate the needed time?

**Comment by MICHAELMUNAVU83** [comment]
It will take me the whole of this week for the whole migration from what I could tell , might be shorter .

**Comment by wintermeyer** [comment]
> It will take me the whole of this week for the whole migration from what I could tell , might be shorter .

Please go for it. We can not postpone this for ever. 

Please try to split it up into smaller issues if possible.

**Comment by MICHAELMUNAVU83** [comment]
Okay , I will focus on this this week

**Comment by MICHAELMUNAVU83** [comment]
@wintermeyer  , I have started on this .
IMO if I push small bits without ensuring all errors/warnings might break the system.
Would be better if I fixed all errors and push it as one PR to at least ensure you can go through the system during the week

**Comment by wintermeyer** [comment]
> IMO if I push small bits without ensuring all errors/warnings might break the system.
Would be better if I fixed all errors and push it as one PR to at least ensure you can go through the system during the week

Your call. I just know from experience that this can become overwhelming pretty fast. But go ahead and tackle it all in one PR.

**Comment by MICHAELMUNAVU83** [comment]
Hello @wintermeyer   , I have made good progress. This was monday after the upgrade

![Screenshot 2024-07-08 at 16 33 24](https://github.com/animina-dating/animina/assets/86654131/e75fdd61-b640-4a86-a0da-0bfb74b04e11)

This is where we are today.
![Screenshot 2024-07-09 at 08 38 29](https://github.com/animina-dating/animina/assets/86654131/091b9743-9937-4d7d-9df9-c0d57c263cfb)


The tests are a real life saver when upgrading versons , I am now going to places that we have not yet written tests for to ensure everything works.

On course to have completed the upgrade by end of week as promised

**Comment by wintermeyer** [comment]
https://elixirforum.com/t/ash-3-1-released/64846

In case it is easy to upgrade to 3.1: Please do so.
